mllib/src/main/scala/org/apache/spark/ml/tuning/ParamGridBuilder.scala

```diff
@@ -100,10 +100,11 @@ class ParamGridBuilder {
    * Builds and returns all combinations of parameters specified by the param grid.
    */
   def build(): Array[ParamMap] = {
-    var paramMaps = Array(new ParamMap)
+    var paramMaps = Array.empty[ParamMap]
```

Comment:
Maybe simpler just to check `values.isEmpty` at the outset?
Not really related, but I don't see why the loop needs an extra `newParamMaps` variable.